### DISCUSSION TITLE
Question about 2FA and email

### DISCUSSION THREAD

**Discussion by Zeex** [community/community#64182] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

I don't want to install a mobile app or use my phone for 2FA. Why is there no option to receive the security codes via primary email? Is it somehow less secure or something? I don't see this being explained in the documentation.

**Comment by arshil121** [comment]
it is possible that the service provider you are using has not implemented email-based 2FA as an option. It is also possible that the service provider has deemed email-based 2FA less secure than other options and has chosen not to offer it. However, it is important to note that any 2FA is better than no 2FA at all

.
If you do not want to install a mobile app or use your phone for 2FA, there are other options available. Authenticator apps are another approach to 2FA second level authentication and have proven to be more secure, more reliable, and faster as well

. Authenticator apps are more secure than email-based 2FA because only the device on which you have the app can provide the code, whereas email is accessible from any device with an internet connection

**Reply by Harem-Rose** [reply]
> only the device on which you have the app can provide the code

Let's explore that for a second.
If my cellular device gets stolen now, without 2FA, can I still access my account? Yes
If my cellular device gets stole, with the forced 2FA, can I still access my account? No

Forced 2FA does not make my account safer from harm, in fact, it increases my threat surface area. Right now, my account is secured by a lengthy password that is unique to this site, not written or saved anywhere, and not used in any password managers. It doesn't exist in a physical sense and cannot be stolen. 
My phone on the other hand, is a very real, very physical, very able-to-be-stolen/Lost/Destroyed object. 

I already have to use my email to verify myself when I log in. That's a 2nd factor already. My email, like my GitHub account, uses a lengthy password that is unique to the email site, is not written or saved anywhere, and is not used in any password managers. It doesn't exist in a physical sense and cannot be stolen.

2FA should be an option that is available, but not forced.

**Reply by nomadbyte** [reply]
> ...My email, like my GitHub account, uses a lengthy password that is unique to the email site, is not written or saved anywhere, and is not used in any password managers.

My understanding is that most online services (GitHub, email providers included) have had security breaches with user personal data stolen. 2FA is their answer to "better security" which is demanded from them by those who got breached with the stolen info.

Does it improve chances of the user info not being stolen again? I doubt that. It mostly depends on the providers own security diligence.

Does it make it more difficult to use the stolen user data for more breaches? It does. Because it adds another auth vector. Though the bad actors can make use of the additional auth detail too, for example, phone numbers.

I guess, the 2FA is a more of a public face policy to show that "we take the issue with your stolen PI data seriously", so you can trust us more of your PI details. Just until it's "Oops, we did it again....".

**Reply by Harem-Rose** [reply]
There may be a misunderstanding on how concerned I am about another person illegitimately accessing my account. While it IS a valid concern, it is not my primary issue.
What I am concerned with is this additional requirement making it difficult, or potentially impossible for me to access my own account. 

2FA massively increases the threat of me being locked out of my own account when trying to access it legitimately. 
2FA for "public face policy" should be offered, but not forced. I am hoping that GitHub sees enough complaints about this forced requirement to make it optional, but I'm not holding my breath. I will have to pull chocks if they do not alter their decision by 25 September. I will not be locked out of my own code.

If GitHub wants to enact "better security" they should be shoring up their internals and not forcing security requirements on the users. This is the pollution equivalent of large companies urging us to drive less and set our thermostats to 80° when they are responsible for 70-80% of the load.

**Comment by Fish-Git** [comment]
> 2FA should be an option that is available, but not forced.

AMEN! This 2FA crap is a PITA and forcing it on users is mostly to protect the service provider (GitHub) from liability when they get hacked and their customer/user data (including password hashes) gets stolen.

Yes, it also helps protects most users too, since most users are idiots and fail to use unique strong passwords for all of their many web site accounts. Instead, they use the same short easy to remember weak password for most all of their many web accounts, so if their password is compromised on one site that doesn't practice good security (and there are many!), it can then be used to access that idiot's many other accounts that foolishly use the same password. They're idiots.

But not all of us are idiots, and I find it extremely insulting for sites such as GitHub to treat us as such.

There is nothing insecure with using a simple userid and password to login to a web site if your userid and password are strong and secure. I have hundreds of accounts and ALL of them use UNIQUE passwords with at least 100+ bits of entropy:

  * https://www.omnicalculator.com/other/password-entropy
  * https://i.imgur.com/gfYw57t.png

My master password which protects my password safe has over 200+ bits of entropy.

I'm not vulnerable to social engineering, phishing, password reset nor even wardriving attacks, as I don't use wireless. I don't visit dangerous web sites (I'm a programmer, not a pervert), I read all my email as plain text. Why do I need 2FA?

Answer: I don't!

Yes, 2FA, if done correctly, is MORE secure than not using it.

But that DOESN'T necessarily mean that NOT using it isn't secure. It's just MORE secure.

But it also comes at a price that I feel I shouldn't be FORCED to have to pay.
